src/components/sectionListRe.rei

```diff
@@ -1,68 +1,76 @@
-type section 'item =
-  Js.t {
-    .
-    data : array 'item, key : Js.Undefined.t string, renderItem : Js.Undefined.t (renderItem 'item)
-  }
-and renderItem 'item =
-  Js.t {
-    .
-    item : 'item,
-    index : int,
-    section : section 'item,
-    separators :
+module Make:
+  (Item: {type item;}) =>
+  {
+    type section =
```

Comment:
Good point, I changed it to a record. But we should keep in mind that this induces some performance penalties (but probably this is more a concern further down the line).

The `viewToken` is too edge case that I wanted to transform it to a record...